### DISCUSSION TITLE
Capturing message header of MQ messages.

### DISCUSSION THREAD

**Discussion by ccoqueiro** [open-telemetry/opentelemetry-specification#2477] [Q&A]
Hey folks!

I saw that we were able to capture all the information that is in the HTTP header in a very simple way, and we capture some by default, and we can add more if we want to add parameters at jvm startup. Now in case of using a messaging system like IBM MQ, fantastically the java agent is able to monitor both the producer and the consumer and also already read some information by default, but I could not identify if it is possible to do as the case of HTTP, I saw that there is a way to capture  consumer receive message,  but then a question, it’s possible add in a tag the information contained in consumer message?

Thanks

**Comment by trask** [comment]
hi @ccoqueiro! [capturing http headers is specified](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers), but there's nothing specified (yet) about capturing messaging headers. if you open a discussion or issue over in https://github.com/open-telemetry/opentelemetry-java-instrumentation we can help you further with currently available options.
